auth/pat.go

```diff
@@ -95,592 +95,188 @@ func ParseOperationType(ot string) (OperationType, error) {
-		}
-	}
-
-	os, exists := ds.Entities[domainEntityType]
-	if !exists {
-		return nil
-	}
-
-	if err := os.Delete(operation, entityIDs...); err != nil {
-		return fmt.Errorf("failed to delete domain %s scope: %w", domainEntityType.String(), err)
-	}
-
-	if len(os) == 0 {
-		delete(ds.Entities, domainEntityType)
-	}
-	return nil
-}
-
-// Check entry in Domain scope.
-func (ds *DomainScope) Check(domainEntityType DomainEntityType, operation OperationType, ids ...string) bool {
-	if ds.Entities == nil {
-		return false
-	}
-	if domainEntityType < DomainManagementScope || domainEntityType > DomainClientsScope {
-		return false
-	}
-	if domainEntityType == DomainManagementScope {
-		return ds.DomainManagement.Check(operation, ids...)
-	}
-	os, exists := ds.Entities[domainEntityType]
-	if !exists {
-		return false
-	}
-
-	return os.Check(operation, ids...)
+func (et *EntityType) UnmarshalText(data []byte) (err error) {
+	str := strings.Trim(string(data), "\"")
+	*et, err = ParseEntityType(str)
+	return err
 }
 
 // Example Scope as JSON
 //
-//	{
-//	    "users": {
-//	            "create": ["*"],
-//	            "read": ["*"],
-//	            "list": ["*"],
-//	            "update": ["*"],
-//	            "delete": ["*"]
-//	    },
-//	    "domains": {
-//	        "domain_1": {
-//	            "entities": {
-//	                "groups": {
-//	                    "create": ["*"] // this for all groups in domain
-//	                },
-//	                "channels": {
-//	                    // for particular channel in domain
-//	                    "delete": [
-//	                        "channel1",
-//	                        "channel2"
-//	                    ]
-//	                },
-//	                "things": {
-//	                    "update": ["*"] // this for all things in domain
-//	                }
-//	            }
-//	        }
-//	    }
-//	}
+// [
+//     {
+//         "optional_domain_id": "domain_1",
+//         "entity_type": "groups",
+//         "operation": "create",
+//         "entity_id": "*"
+//     },
+//     {
+//         "optional_domain_id": "domain_1",
+//         "entity_type": "channels",
+//         "operation": "delete",
+//         "entity_id": "channel1"
+//     },
+//     {
+//         "optional_domain_id": "domain_1",
+//         "entity_type": "things",
+//         "operation": "update",
+//         "entity_id": "*"
+//     }
+// ]
+
 type Scope struct {
-	Users     OperationScope         `json:"users,omitempty"`
-	Domains   map[string]DomainScope `json:"domains,omitempty"`
-	Dashboard OperationScope         `json:"dashboard,omitempty"`
-	Messaging OperationScope         `json:"messaging,omitempty"`
+	PatId            string
+	OptionalDomainId string     `json:"optional_domain_id,omitempty"`
+	EntityType       EntityType `json:"entity_type,omitempty"`
+	EntityId         string     `json:"entity_id,omitempty"`
+	Operation        Operation  `json:"operation,omitempty"`
```

Comment:
Why do we have a scope this complex? Can we have it embedded entity types and operations only?